src/main/java/cz/it4i/parallel/ImageJServerWorker.java

```diff
@@ -62,191 +59,160 @@ public String getHostName() {
-			httpPost.setEntity(entity);
-			HttpResponse response = httpClient.execute(httpPost);
-			json = EntityUtils.toString(response.getEntity());
+			String postUrl = "http://" + hostName + ":" + String.valueOf(port) + "/objects/upload";
+			HttpPost httpPost = new HttpPost(postUrl);
 
-			// TODO check result code
+			HttpEntity entity = MultipartEntityBuilder.create()
+					.addBinaryBody("file", new File(filePath), ContentType.create(getContentType(filePath)), fileName)
+					.build();
+			httpPost.setEntity(entity);
 
+			HttpResponse response = HttpClientBuilder.create().build().execute(httpPost);
+
+			// TODO check result code properly
+
+			String json = EntityUtils.toString(response.getEntity());
+			String obj = new org.json.JSONObject(json).getString("id");
+			
+			result = Mockito.mock(Dataset.class, (Answer<Dataset>) p -> {
+				throw new UnsupportedOperationException();
+			});
+			doAnswer(p -> "Dataset(mocked)[id = " + obj).when(result).toString();
+			
+			mockedData2id.put(result, obj);
+			id2mockedData.put(obj, result);
+			
 		} catch (Exception e) {
 			e.printStackTrace();
 		}
 
-		String obj = new org.json.JSONObject(json).getString("id");
-		Dataset result = Mockito.mock(Dataset.class, (Answer<Dataset>) p -> {
-			throw new UnsupportedOperationException();
-		});
-		doAnswer(p -> "Dataset(mocked)[id = " + obj).when(result).toString();
-		
-		mockedData2id.put(result, obj);
-		id2mockedData.put(obj, result);
 		return result;
 	}
 
+	@Override
 	public void exportData(Dataset dataset, Path p) {
-		String filePath = p.toString();
-		String id = mockedData2id.get(dataset);
-		String getUrl = "http://" + hostName + ":" + String.valueOf(port) + "/objects/" + id + "/" + getImageType(filePath);
-		HttpClient httpClient = HttpClientBuilder.create().build();
-		HttpGet get = new HttpGet(getUrl);
+
+		final String filePath = p.toString();
+		final String objectId = mockedData2id.get(dataset);
 
 		try {
 
-			HttpResponse response = httpClient.execute(get);
-			HttpEntity entity = response.getEntity();
+			String getUrl = "http://" + hostName + ":" + String.valueOf(port) + "/objects/" + objectId + "/"
+					+ getImageType(filePath);
+			HttpGet httpGet = new HttpGet(getUrl);
+
+			HttpEntity entity = HttpClientBuilder.create().build().execute(httpGet).getEntity();
+
 			if (entity != null) {
-				// long len = entity.getContentLength();
-				BufferedInputStream bis = new BufferedInputStream(entity.getContent());
-				BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(new File(filePath)));
-				int inByte;
-				while ((inByte = bis.read()) != -1) {
-					bos.write(inByte);
-				}
-				bis.close();
-				bos.close();
 
-			}
+				BufferedInputStream bis = null;
+				BufferedOutputStream bos = null;
 
-		} catch (Exception e) {
-			e.printStackTrace();
-		}
+				try {
 
-	}
+					bis = new BufferedInputStream(entity.getContent());
+					bos = new BufferedOutputStream(new FileOutputStream(new File(filePath)));
```

Comment:
It could be placed into "try" header for autoclose.

Reply:
Done.